Multiple tests mixing up Python version between test venv and system

### Describe the bug

When running the test suite in a virtual environment where the Python version used in the virtual environment is different from the system Python, it seems that the fixtures use the wrong (system) Python version rather than the version used in the virtual environment (and expected by the tests).

For example, when my `/usr/bin/python` is 3.13, but virtualenv is made using Python 3.12, I'm getting:

```pytb
_____________________________________________________ test_project_packages_path ______________________________________________________
[gw8] linux -- Python 3.12.10 /tmp/pdm/.venv/bin/python

project = <Project '/tmp/pytest-of-mgorny/pytest-12/popen-gw8/test_project_packages_path0'>

    def test_project_packages_path(project):
        packages_path = project.environment.packages_path
        version = ".".join(map(str, sys.version_info[:2]))
        if os.name == "nt" and sys.maxsize <= 2**32:
            assert packages_path.name == version + "-32"
        else:
>           assert packages_path.name == version
E           AssertionError: assert '3.13' == '3.12'
E             
E             - 3.12
E             ?    ^
E             + 3.13
E             ?    ^

tests/test_project.py:149: AssertionError
-------------------------------------------------------- Captured stdout setup --------------------------------------------------------
Changes are written to pyproject.toml.
```

### To reproduce

```console
$ git rev-parse HEAD
ba2af395c27cd378b326a7ce22244d755e11e580
$ python --version
Python 3.13.3
$ pdm venv create 3.12
Virtualenv /tmp/pdm/.venv is created successfully
$ pdm install -dGtest
INFO: Virtualenv /tmp/pdm/.venv is reused.
Synchronizing working set with resolved packages: 50 to add, 0 to update, 0 to remove

  ✔ Install findpython 0.6.3 successful
  ✔ Install id 1.5.0 successful
  ✔ Install blinker 1.8.2 successful
  ✔ Install certifi 2024.8.30 successful
  ✔ Install filelock 3.16.1 successful
  ✔ Install h11 0.14.0 successful
  ✔ Install dep-logic 0.5.0 successful
  ✔ Install colorama 0.4.6 successful
  ✔ Install execnet 2.1.1 successful
  ✔ Install hishel 0.0.33 successful
  ✔ Install idna 3.6 successful
  ✔ Install httpx 0.27.2 successful
  ✔ Install anyio 4.3.0 successful
  ✔ Install httpcore 1.0.6 successful
  ✔ Install distlib 0.3.8 successful
  ✔ Install iniconfig 2.0.0 successful
  ✔ Install mdurl 0.1.2 successful
  ✔ Install pluggy 1.5.0 successful
  ✔ Install platformdirs 4.3.6 successful
  ✔ Install pyproject-hooks 1.2.0 successful
  ✔ Install packaging 24.1 successful
  ✔ Install installer 0.7.0 successful
  ✔ Install pbs-installer 2025.2.12 successful
  ✔ Install pytest-httpserver 1.1.0 successful
  ✔ Install pytest-httpx 0.34.0 successful
  ✔ Install pytest-cov 5.0.0 successful
  ✔ Install pytest-mock 3.14.0 successful
  ✔ Install pytest-rerunfailures 14.0 successful
  ✔ Install python-dotenv 1.0.1 successful
  ✔ Install pytest-xdist 3.6.1 successful
  ✔ Install resolvelib 1.1.0 successful
  ✔ Install sniffio 1.3.1 successful
  ✔ Install socksio 1.0.0 successful
  ✔ Install markdown-it-py 3.0.0 successful
  ✔ Install requests 2.31.0 successful
  ✔ Install tomlkit 0.13.2 successful
  ✔ Install pytest 8.3.3 successful
  ✔ Install truststore 0.9.2 successful
  ✔ Install unearth 0.17.5 successful
  ✔ Install typing-extensions 4.12.2 successful
  ✔ Install rich 13.9.2 successful
  ✔ Install shellingham 1.5.4 successful
  ✔ Install urllib3 2.2.1 successful
  ✔ Install werkzeug 2.2.3 successful
  ✔ Install virtualenv 20.27.0 successful
  ✔ Install pygments 2.17.2 successful
  ✔ Install markupsafe 2.1.5 successful
  ✔ Install charset-normalizer 3.3.2 successful
  ✔ Install msgpack 1.1.0 successful
  ✔ Install coverage 7.2.7 successful
  ✔ Install pdm 2.24.0 successful

  0:00:04 🎉 All complete! 50/50
$ pdm run pytest -n12 --dist=worksteal
========================================================= test session starts =========================================================
platform linux -- Python 3.12.10, pytest-8.3.3, pluggy-1.5.0
rootdir: /tmp/pdm
configfile: pyproject.toml
testpaths: tests/
plugins: xdist-3.6.1, rerunfailures-14.0, mock-3.14.0, cov-5.0.0, pytest_httpserver-1.1.0, httpx-0.34.0, anyio-4.3.0
12 workers [1017 items]
[…]
======================================================= short test summary info =======================================================
SKIPPED [1] tests/test_utils.py:170: Windows test
XFAIL tests/cli/test_others.py::test_search_package - HTTP search risk controlled
FAILED tests/cli/test_hooks.py::test_skip_pre_post_option_from_signal[post-publish] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=:post'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw3/test_skip_pre_post_option_from1/logs/pdm-build-5inf4m3b.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/resolver/test_resolve.py::test_resolve_local_artifacts[sdist] - resolvelib.resolvers.exceptions.ResolutionImpossible: [RequirementInformation(requirement=NamedRequirement(name='setuptools', marker=None, extras=set(), specifier=<SpecifierSet('>=61')>, editable=False, prerelease=None, groups=['default']), parent=None)]
FAILED tests/resolver/test_resolve.py::test_resolve_two_extras_from_the_same_package - resolvelib.resolvers.exceptions.ResolutionImpossible: [RequirementInformation(requirement=NamedRequirement(name='setuptools', marker=None, extras=set(), specifier=<SpecifierSet('>=61')>, editable=False, prerelease=None, groups=['default']), parent=None)]
FAILED tests/cli/test_init.py::test_new_command - AssertionError: assert {'project': {...ion': False}}} == {'project': {...ion': False}}}
  
  Omitting 1 identical items, use -vv to show
  Differing items:
  {'project': {'authors': [{'email': 'me@example.org', 'name': 'Testing'}], 'dependencies': [], 'description': 'Default template for PDM package', 'license': {'text': 'MIT'}, ...}} != {'project': {'authors': [{'email': 'me@example.org', 'name': 'Testing'}], 'dependencies': [], 'description': 'Default template for PDM package', 'license': {'text': 'MIT'}, ...}}
  
  Full diff:
    {
        'project': {
            'authors': [
                {
                    'email': 'me@example.org',
                    'name': 'Testing',
                },
            ],
            'dependencies': [],
            'description': 'Default template for PDM package',
            'license': {
                'text': 'MIT',
            },
            'name': 'test-project',
            'readme': 'README.md',
  -         'requires-python': '==3.13.*',
  ?                                  ^
  +         'requires-python': '==3.12.*',
  ?                                  ^
            'version': '0.1.0',
        },
        'tool': {
            'pdm': {
                'distribution': False,
            },
        },
    }
FAILED tests/resolver/test_resolve.py::test_resolve_file_req_with_prerelease - resolvelib.resolvers.exceptions.ResolutionImpossible: [RequirementInformation(requirement=NamedRequirement(name='setuptools', marker=None, extras=set(), specifier=<SpecifierSet('>=61')>, editable=False, prerelease=None, groups=['default']), parent=None)]
FAILED tests/cli/test_hooks.py::test_hooks[build] - RuntimeError: Call command ['build'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_hooks_build_0/logs/pdm-build-1g9733ex.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_venv.py::test_conda_backend_create[True] - AssertionError: assert False
 +  where False = <built-in method startswith of str object at 0x7f207f5a5170>('3.12')
 +    where <built-in method startswith of str object at 0x7f207f5a5170> = '3.13'.startswith
 +      where '3.13' = <pdm.cli.commands.venv.backends.CondaBackend object at 0x7f207f5a6db0>.ident
FAILED tests/cli/test_venv.py::test_conda_backend_create[False] - AssertionError: assert False
 +  where False = <built-in method startswith of str object at 0x7f207f85e430>('3.12')
 +    where <built-in method startswith of str object at 0x7f207f85e430> = '3.13'.startswith
 +      where '3.13' = <pdm.cli.commands.venv.backends.CondaBackend object at 0x7f207f85c470>.ident
FAILED tests/cli/test_hooks.py::test_hooks[publish] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_hooks_publish_0/logs/pdm-build-udfr4y4q.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[build-pre_build] - RuntimeError: Call command ['build', '--skip=pre_build'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_b0/logs/pdm-build-zlrmmv42.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[build-post_build] - RuntimeError: Call command ['build', '--skip=post_build'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_b1/logs/pdm-build-foc1d7hb.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[publish-pre_publish] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=pre_publish'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_p0/logs/pdm-build-0zfpxpus.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[publish-pre_build] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=pre_build'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_p1/logs/pdm-build-qxr43jd0.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[publish-post_build] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=post_build'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_p2/logs/pdm-build-gj6wreyr.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_option_from_signal[publish-post_publish] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=post_publish'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_option_from_signal_p3/logs/pdm-build-l4wqeggc.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_build.py::test_build_with_no_isolation - AssertionError: assert 0 == 1
 +  where 0 = RunResult(exit_code=0, stdout='Building sdist...\nBuilt sdist at /tmp/pytest-of-mgorny/pytest-12/popen-gw0/test_build_...test-12/popen-gw0/test_build_with_no_isolation0/dist/test_project-0.0.0-py3-none-any.whl\n', stderr='', exception=None).exit_code
FAILED tests/cli/test_hooks.py::test_skip_all_option_from_signal[:all-build] - RuntimeError: Call command ['build', '--skip=:all'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_all_option_from_sign1/logs/pdm-build-k0dr72jn.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_all_option_from_signal[:all-publish] - RuntimeError: Call command ['publish', '--username', 'abc', '--password', '123', '--skip=:all'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_all_option_from_sign5/logs/pdm-build-mkug73yo.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/cli/test_hooks.py::test_skip_all_option_from_signal[:pre,:post-build] - RuntimeError: Call command ['build', '--skip=:pre,:post'] failed(1): See /tmp/pytest-of-mgorny/pytest-12/popen-gw1/test_skip_all_option_from_sign11/logs/pdm-build-k3_clgg5.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='pdm-backend', marker=None, extras=set(), specifier=<SpecifierSet('')>, editable=False, prerelease=None, groups=['default']), parent=None)]
WARNING: Add '-v' to see the detailed traceback
FAILED tests/test_project.py::test_project_packages_path - AssertionError: assert '3.13' == '3.12'
  
  - 3.12
  ?    ^
  + 3.13
  ?    ^
================================= 20 failed, 995 passed, 1 skipped, 1 xfailed, 425 warnings in 38.06s =================================
```

### Expected Behavior

The test suite using a consistent Python version.

### Environment Information

```console
$ pdm info
PDM version:
  2.24.0
Python Interpreter:
  /tmp/pdm/.venv/bin/python (3.12)
Project Root:
  /tmp/pdm
Local Packages:
  
$ pdm info --env
{
  "implementation_name": "cpython",
  "implementation_version": "3.12.10",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "6.14.2-gentoo-dist",
  "platform_system": "Linux",
  "platform_version": "#1 SMP PREEMPT_DYNAMIC Thu Apr 10 14:55:39 -00 2025",
  "python_full_version": "3.12.10",
  "platform_python_implementation": "CPython",
  "python_version": "3.12",
  "sys_platform": "linux"
}
```

### Verbose Command Output

_No response_

### Additional Context

We're hitting the same problem while trying to package PDM for Gentoo Linux, though (as noted above) I can reproduce with pure `pdm` venv.

Not sure if I should check the checkbox below — I'd be happy to submit a pull request if you told me what needs to be done, because I have no clue where the problem lies.

### Are you willing to submit a PR to fix this bug?

- [x] Yes, I would like to submit a PR.

--- a/src/pdm/pytest.py
+++ b/src/pdm/pytest.py
@@ -63,7 +63,7 @@ from pdm.models.requirements import (
 from pdm.models.session import PDMPyPIClient
 from pdm.project.config import Config
 from pdm.project.core import Project
-from pdm.utils import find_python_in_path, normalize_name, parse_version
+from pdm.utils import normalize_name, parse_version
 
 if TYPE_CHECKING:
     from typing import Protocol
@@ -407,9 +407,7 @@ def project_no_init(
     tmp_path.joinpath("caches").mkdir(parents=True)
     p.global_config["cache_dir"] = tmp_path.joinpath("caches").as_posix()
     p.global_config["log_dir"] = tmp_path.joinpath("logs").as_posix()
-    python_path = find_python_in_path(sys.base_prefix)
-    if python_path is None:
-        raise ValueError("Unable to find a Python path")
+    python_path = Path(sys.executable)
     p._saved_python = python_path.as_posix()
     monkeypatch.delenv("VIRTUAL_ENV", raising=False)
     monkeypatch.delenv("CONDA_PREFIX", raising=False)

